subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/RealWorldNativePluginPerformanceTest.groovy

```diff
@@ -73,43 +71,37 @@ class RealWorldNativePluginPerformanceTest extends AbstractCrossVersionGradleInt
 
         def changedFile = fileToChange
         def changeClosure = change
-        runner.addBuildExperimentListener(new BuildExperimentListenerAdapter() {
-            String originalContent
-            File originalContentFor
-
-            @Override
-            void beforeInvocation(BuildExperimentInvocationInfo invocationInfo) {
-                File file = new File(invocationInfo.projectDir, changedFile)
-                if (originalContentFor != file) {
-                    assert file.exists()
-                    def backupFile = new File(file.parentFile, file.name + "~")
-                    if (backupFile.exists()) {
-                        originalContent = backupFile.text
+        runner.addBuildMutator { invocationSettings ->
```

Comment:
I think Gradle profiler already has mutators for header and source files. Can we re-use those?

I also think the `build file` change should go to a different test and not modify the build file but pick up the compiler defines by a Gradle property or maybe an external file. WDYT?

Reply:
You mean `ApplyChangeToNativeSourceFileMutator`, IMHO that class can't help much in this case, because this test has 3 different kind of mutations. Regarding the `build file` change, I'd put it into followups.

Reply:
2 of the three kinds are covered by `ApplyChangeToNativeSourceFileMutator`. It is fine like it is now, though.